src/vcpkg/commands.integrate.cpp

```diff
@@ -14,6 +14,101 @@
 
 namespace vcpkg::Commands::Integrate
 {
+    Optional<int> find_targets_file_version(StringView contents)
+    {
+        constexpr static StringLiteral VERSION_START = "<!-- version ";
+        constexpr static StringLiteral VERSION_END = " -->";
+
+        auto first = contents.begin();
+        const auto last = contents.end();
+        for (;;)
+        {
+            first = std::search(first, last, VERSION_START.begin(), VERSION_START.end());
+            if (first == last)
+            {
+                break;
+            }
+            first += VERSION_START.size();
+            auto version_end = std::search(first, last, VERSION_END.begin(), VERSION_END.end());
+            if (version_end == last)
+            {
+                break;
+            }
+            auto ver = Strings::strto<int>({first, version_end});
+            if (ver.has_value() && *ver.get() >= 0)
+            {
+                return ver;
+            }
+        }
+        return nullopt;
+    }
+
+    std::vector<std::string> get_bash_source_completion_lines(StringView contents)
+    {
+        std::vector<std::string> matches;
+        auto first = contents.begin();
+        const auto last = contents.end();
+        while (first != last)
+        {
+            const auto end_of_line = std::find(first, last, '\n');
+            const auto line = Strings::trim(StringView{first, end_of_line});
+            if (Strings::starts_with(line, "source") && Strings::ends_with(line, "scripts/vcpkg_completion.bash"))
+            {
+                matches.emplace_back(line.begin(), line.end());
```

Comment:
```suggestion
                matches.emplace_back(line.data(), line.size());
```

The char*,size_t constructor for string is slightly more efficient because it doesn't need metaprogramming to determine that the input is a contiguous range.